core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,310 @@
+                        break;
+                    case REQUEST:
+                        parameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, ++position)), String.format("Content-Type, %s", value)));
+                        break;
+                    case RESPONSE:
+                        parameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, ++position)), String.format("Accept, %s", value)));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        for (Parameter parameter : operation.getParameters()) {
+            String parameterName = parameter.getName();
+            if (parameter.getIn().equals("header") && !partNames.containsValue(parameterName)) {
+                parameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, ++position)), String.format("%s, #%s#",parameterName,parameterName)));
+            }
+        }
+
+        return parameters;
+    }
+
+    public void addSecurityHeader(SecurityScheme securityScheme, List<ComponentParameter> componentParameters,String componentName, OpenAPI openAPI, int position, String value) {
+        long versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        String scheme = securityScheme.getScheme();
+        SecurityScheme.Type securityType = securityScheme.getType();
+        
+
+        if (securityType == null) {
+            log.warn("The securityScheme provided doesn't exists");
+            return;
+        }
+        if (securityType.equals(SecurityScheme.Type.OAUTH2) || (securityType.equals(SecurityScheme.Type.HTTP) && scheme.equals("bearer"))) {
+            componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, position)), String.format("Authorization, Bearer #%s#", value)));
+        }
+        else if (securityScheme.getType().equals(SecurityScheme.Type.HTTP) && scheme.equals("basic" )) {
+            componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, position)), String.format("Authorization, Basic #%s#", value)));
+        }
+        else if (securityScheme.getType().equals(SecurityScheme.Type.APIKEY)) {
+            componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, position)), String.format("X-API-KEY, #%s#", value)));
+        } else if (securityScheme.getType().equals(SecurityScheme.Type.OPENIDCONNECT)) {
+            componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, versionNumber, String.format(HEADER, position)), String.format("X-API-KEY, #%s#", value)));
+        }
+
+    }
+
+    public String translatePathName(String pathName) {
```

Comment:
I will remove this method for the next push, because I am using this replacement once